kafka/producer.go

```diff
@@ -360,6 +360,21 @@ func (p *Producer) Close() {
 	C.rd_kafka_destroy(p.handle.rk)
 }
 
+// Purge messages from queue. If purgeInFlight is true, then the inflight message will be purged as well.
+func (p *Producer) Purge(purgeInFlight bool) error {
```

Comment:
For full flexibility the current flags, and any future flags, needs to be supported.
So make this a flag field.